### PR TITLE
Get gtest 1.10.0 and parametrize tests

### PR DESCRIPTION
**Pull request by antalszava**
**Changes**

* Changes to CI to get the latest release of Google test available on GitHub (`v1.10.0`), [see release notes](https://github.com/google/googletest/releases/tag/release-1.10.0)
* Creates parametrized tests

### CODE REVIEW

File: pennylane_lightning/src/tests/lightning_util_unittest.cpp [hunk 1]
```diff
@@ -14,33 +14,47 @@
...
+class Exp2TestFixture :public ::testing::TestWithParam<std::tuple<int, int>> {
+};
 
-    for (size_t i = 0; i < inputs.size(); i++) {
-        size_t result = Pennylane::exp2(inputs[i]);
-        EXPECT_TRUE(result == outputs[i]);
-    }
+TEST_P(Exp2TestFixture, CheckExp2Results) {
+    int input = std::get<0>(GetParam());
+    int expected = std::get<1>(GetParam());
```
Comment (Thenerdstation) [inline pennylane_lightning/src/tests/lightning_util_unittest.cpp:26]: `std::get<0>(GetParam()):` is the python equivalent of `a[0]` lol

**Review by Thenerdstation — APPROVED** [review]

File: pennylane_lightning/src/tests/lightning_util_unittest.cpp (hunk 1 above)
Comment (trbromley) [inline pennylane_lightning/src/tests/lightning_util_unittest.cpp:26]: Do you get any warnings when using an `int` type? For me, it was giving a warning of comparing `int` and `size_t`.

**Review by trbromley — APPROVED** [review]
Thanks @antalszava, looks great!

I'm still not convinced of the benefits of this approach versus the "good old for loop", since for me there is also a cost to this approach of more complexity and less understandable code.

But maybe where this approach shines is in more advanced test cases. I'm good to go with this approach for now.

File: pennylane_lightning/src/tests/lightning_util_unittest.cpp (hunk 1 above)
Comment (antalszava) [reply pennylane_lightning/src/tests/lightning_util_unittest.cpp:26]: Yep, same here, warnings are emitted there.

**Comment by antalszava** [comment]
@trbromley agreed that a `for` loop is actually briefer in these cases! One advantage with parameterized test cases is that they can be easier to parse on test failure as the cases are separated based on parameters. Though in the cases of this PR, I agree that there is not such a big difference.

**Pull request merged**